cmd/kubeadm/app/cmd/phases/init/waitcontrolplane.go

```diff
@@ -58,17 +59,25 @@ var (
 // NewWaitControlPlanePhase is a hidden phase that runs after the control-plane and etcd phases
 func NewWaitControlPlanePhase() workflow.Phase {
 	phase := workflow.Phase{
-		Name:   "wait-control-plane",
-		Run:    runWaitControlPlanePhase,
-		Hidden: true,
+		Name: "wait-control-plane",
+		Phases: []workflow.Phase{
+			{
+				Name:   "wait-kubelet",
+				Run:    runWaitKubeletPhase,
+				Hidden: true,
+			},
+			newWaitControlPlaneComponentSubphase(kubeadmconstants.KubeAPIServer, true),
+			newWaitControlPlaneComponentSubphase(kubeadmconstants.KubeControllerManager, true),
+			newWaitControlPlaneComponentSubphase(kubeadmconstants.KubeScheduler, false),
+		},
 	}
 	return phase
 }
 
-func runWaitControlPlanePhase(c workflow.RunData) error {
+func runWaitKubeletPhase(c workflow.RunData) error {
```

Comment:
Can you just move the logic from `newWaitControlPlaneComponentSubphase` to this method `runWaitControlPlanePhase`? and guard it with a FG?